src/goTest.ts

```diff
@@ -32,8 +32,12 @@ export function testAtCursor(goConfig: vscode.WorkspaceConfiguration, args: any)
 		return;
 	}
 	if (editor.document.isDirty) {
-		vscode.window.showInformationMessage('File has unsaved changes. Save and try again.');
-		return;
+		if (goConfig['saveTestFileOnRunOrDebug'] === true) {
+			editor.document.save();
```

Comment:
I believe, `editor.document.save()` returns a promise. We should wait for the promise to complete before trying to run the tests.

Reply:
👍 thanks! Will take a stab (it's been a _long_ time since I've used promises and this is my first `typescript` PR, so thanks for your patience) and re-submit.